storey/sources.py

```diff
@@ -730,212 +759,80 @@ def __init__(
-                            if create_event:
-                                if self._id_field:
-                                    id_field = self._id_field
-                                    if self._with_header and isinstance(id_field, str):
-                                        id_field = field_name_to_index[id_field]
-                                    id = parsed_line[id_field]
-                                else:
-                                    id = self._get_uuid()
-                                event = Event(element, key=key, id=id)
-                                self._event_buffer.put(event)
-                            else:
-                                if self.context:
-                                    self.context.logger.error(
-                                        f"For {parsed_line} value of key {single_key_field} is None"
-                                    )
-                        except BaseException as ex:
-                            ex._raised_by_storey_step = self
-                            if self.context and hasattr(self.context, "push_error"):
-                                message = traceback.format_exc()
-                                if self.logger:
-                                    self.logger.error(f"Pushing error to error stream: {ex}\n{message}")
-                                if not event:
-                                    if parsed_line is not None:
-                                        event = Event(parsed_line)
-                                    else:
-                                        event = Event(line)
-                                self.context.push_error(event, f"{ex}\n{message}", source=self.name)
-                            else:
-                                raise ex
-                if self._with_header:
-                    self._dates_indices = []
-
-        except BaseException as ex:
-            self._event_buffer.put(ex)
-        self._event_buffer.put(_termination_obj)
-
-    def _get_event(self):
-        event = self._event_buffer.get()
-        if isinstance(event, BaseException):
-            raise event
-        return event
+    def _get_element(self, body: OrderedDict):
+        if self._build_dict and self._with_header:
+            return dict(body)
+        return list(body.values())
 
-    async def _run_loop(self):
-        asyncio.get_running_loop().run_in_executor(None, self._blocking_io_loop)
-
-        def get_multiple():
-            events = [self._get_event()]
-            while not self._event_buffer.empty() and len(events) < 128:
-                events.append(self._get_event())
-            return events
+    def _get_by_field_or_index(self, field, body: OrderedDict):
+        if self._with_header and isinstance(field, str):
+            result = super()._get_by_field_or_index(field=field, body=body)
+        else:
+            result = list(body.items())[field][1]
```

Comment:
I'm not sure how we ended up accessing a dictionary by position, but if it does make sense,
```suggestion
            result = list(body.values())[field]
```